src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -879,24 +931,96 @@ void NativeORCBlockInputFormat::prepareFileReader()
     }
     include_indices.assign(include_typeids.begin(), include_typeids.end());
 
-    if (format_settings.orc.filter_push_down && key_condition && !sarg)
+    if (format_settings.orc.filter_push_down && key_condition && !sargs)
+    {
+        sargs = buildORCSearchArgument(*key_condition, getPort().getHeader(), file_reader->getType(), format_settings);
+        sargs_applier = std::make_shared<orc::SargsApplier>(
+            file_reader->getType(), sargs.get(), file_reader->getRowIndexStride(), file_reader->getWriterVersion(), nullptr);
+    }
+
+    selected_stripes = calculateSelectedStripes();
+    stripes_iterator = 0;
+
+    if (use_prefetch)
+    {
+        /// Prefetch first selected stripe
+        prefetchStripe(stripes_iterator);
```

Comment:
need coalesce small stripes, when strip size is smaller than min_size_for_seek

Reply:
indeed. I'll fix it.